### PR TITLE
gateio/kucoin: assortment of fixes

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Gateio:
* fix unmarshal bug and update fields
* Improves wrapper function for fetching open contracts. The prior one fetched data per contract.
* Adds missing details and handling for order details
* When fetching balances, continue if specific error has been found and warn client of error details.
* Purge generating random string for client order ID when it is not needed for wrapper, this can be set from caller.
* Fix bug on outbound order submit through wrapper

Kucoin:
* Fix bug in get active orders where nothing was being processed if you didn't pass in currency pairs
* Adds missing details and handling for order details
* Fetch all settlement balances
* Add handling for long/short order sides when submitting order
* Remove leverage check as this should be set by caller. 

Currency:
* Add helper method to currency pairs to return a match either base or quote. 

Order:
* Add settlement currency field
* Adds execution note field (Can RM this field if you guys don't want it, it was a carry over from GATEIO field ClientOrderID:   futuresOrders[x].Text,)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by shazbert** [comment]
Problem with market deployment.

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Only minor things right now

File: exchanges/gateio/gateio_wrapper.go [hunk 1]
```diff
@@ -1622,50 +1688,81 @@ func (g *Gateio) GetActiveOrders(ctx context.Context, req *order.MultiOrderReque
...
+					Amount:               futuresOrders[x].Size,
+					ContractAmount:       futuresOrders[x].Size,
+					Pair:                 pair,
+					OrderID:              strconv.FormatInt(futuresOrders[x].ID, 10),
+					Price:                futuresOrders[x].OrderPrice.Float64(),
+					ExecutedAmount:       futuresOrders[x].Size - futuresOrders[x].RemainingAmount,
+					RemainingAmount:      futuresOrders[x].RemainingAmount,
+					LastUpdated:          futuresOrders[x].FinishTime.Time(),
+					Date:                 futuresOrders[x].CreateTime.Time(),
+					ExecutionNote:        futuresOrders[x].Text,
```
[the remark this answers is not in the document]
Comment (shazbert) [reply exchanges/gateio/gateio_wrapper.go:1753]: Yeah this is strange when you don't supply a client order id it comes back `"text":"api",` or front end. When you do it returns the client order ID like expected I will purge this.

File: exchanges/gateio/gateio_wrapper.go [hunk 2]
```diff
@@ -1622,50 +1688,81 @@ func (g *Gateio) GetActiveOrders(ctx context.Context, req *order.MultiOrderReque
...
+					log.Warnf(log.ExchangeSys, "%s %v", g.Name, err)
+					continue
+				}
 				return nil, err
 			}
 			for x := range futuresOrders {
 				if futuresOrders[x].Status != "open" {
 					continue
 				}
+				var pair currency.Pair
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/gateio/gateio_wrapper.go:1701]: Thanks for making those changes! I'm happy with the PR overall, just waiting on this

**Comment by gloriousCode** [comment]
Happy with changes, just need exchange_wrapper_standards addressing: https://github.com/thrasher-corp/gocryptotrader/actions/runs/8074480255/job/22059833368?pr=1404#step:9:292
For Kucoin, the &order.Submit needs leverage there since you now require it to be sent
For gateio, GetActiveOrders now requires a pair, but the error is not something exchange_wrapper_standards can access. Is there a better error to select?

**Review by gloriousCode — APPROVED** [review]
# 🌯  🌞 
tACK!

**Review by thrasher- — CHANGES_REQUESTED** [review]
Lovely! Some minor nitters

**Review by thrasher- — APPROVED** [review]
Thanks for making those changes

**Pull request merged**